src/main/java/org/jenkinsci/test/acceptance/plugins/stageview/StageViewBuild.java

```diff
@@ -56,8 +56,8 @@ public List<StageViewStage> getAllStageViewItem() {
      *
      * @return specific item
      */
-    public StageViewStage getStageViewItem(int buildNumber) {
-        return this.stageViewStages.get(stageViewStages.size() - 1);
+    public StageViewStage getStageViewItem(int stageNumber) {
+        return this.stageViewStages.get(stageNumber);
```

Comment:
this has only ever worked by fluke...  there is a Test that is actually calling this and expecting a different response based on the stage, it would only pass if the last stage failed in under 1 second.

https://github.com/jenkinsci/acceptance-test-harness/blob/0f06fc2dab9247ab45ed40d7764044bc9d370c7b/src/test/java/plugins/StageViewTest.java#L146-L150